### DISCUSSION TITLE
window.livewire.rescan() produces double events

### DISCUSSION THREAD

**Discussion by mikegk** [livewire/livewire#5034] [1. Help]
Hi guys,

I do have got a very strange behave as result of an AJAX request which should render a livewire component.

This is what I do:

- A button is used to trigger the AJAX call
- The Controller returns an rendered embed livewire component (<div>@livewire('some.component')</div>):
`return view('wrapper.component-wrapper')->render()`
-The result is beeing added to another wrapper via JS on successfull post-request

The component contains another component, and this one also contains a last component (3 nested).

After fetching the data, the components are rendered but buttons won't work.
So I used:
`window.livewire.rescan()`
To make them work. The Problem is: After calling rescan(), event is emitted twice afterwards.
I also tried to use window.livewire.start(), restart() and stop() with no success.

I hope you can reproduce the error.

**Comment by joshhanley** [comment]
@mikegk I'm not sure if loading over AJAX is supported, but if you can share your code, maybe we can have a look and see if we can get it working, as I don't understand what you mean by the controller returns an embedded Livewire component.

Also using `rescan` won't remove the old listeners, so that would be why you are getting a double up of events.

Thanks!

**Comment by luttje** [comment]
I found similar problems, and although we're not yet sure if it's intended behavior, you can use this JS to re-bind the events (without doubling them):
```js
// Get the Livewire component your ajax buttons are in
const component = template.closest('[wire\\3A id]')
// Remove all livewire events
component.__livewire.tearDown()
// Re-add all livewire events
compoment.__livewire.initialize()
```
